uvicorn/_subprocess.py

```diff
@@ -18,10 +18,42 @@
 spawn = multiprocessing.get_context("spawn")
 
 
+class SocketSharePickle:
+    def __init__(self, sock: socket.socket):
+        self._sock = sock
+
+    def get(self) -> socket.socket:
+        return self._sock
+
+
+class SocketShareRebind:
+    def __init__(self, sock: socket.socket):
+        if not (sys.platform == "linux" and hasattr(socket, "SO_REUSEPORT")) or hasattr(socket, "SO_REUSEPORT_LB"):
```

Comment:
Wouldn't it be better to put the logic here in `config.bind_socket`?
Just add it to the else, uds and fd don't seem to support SO_REUSEPORT.

Reply:
hmm I'm not sure about this, I want to make sure the SO_REUSEPORT flag isn't set before the socket is bound so we still fail if another uvicorn is using this port

Reply:
How about this?
https://github.com/vvanglro/uvicorn/commit/f2b103712c4fde78fa6f49c9e9f552ca9f30aa18